fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/JDBCIncludeRevincludeTest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (C) Copyright IBM Corp. 2017,2019
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.test;
+
+import java.util.Properties;
+
+import com.ibm.fhir.model.test.TestUtil;
+import com.ibm.fhir.persistence.FHIRPersistence;
+import com.ibm.fhir.persistence.jdbc.impl.FHIRPersistenceJDBCImpl;
+import com.ibm.fhir.persistence.jdbc.test.util.DerbyInitializer;
+import com.ibm.fhir.persistence.test.common.AbstractIncludeRevincludeTest;
+
+
+public class JDBCIncludeRevincludeTest extends AbstractIncludeRevincludeTest {
+    
+    private Properties testProps;
+    
+    public JDBCIncludeRevincludeTest() throws Exception {
+        this.testProps = TestUtil.readTestProperties("test.jdbc.properties");
+    }
+
+    @Override
+    public void bootstrapDatabase() throws Exception {
+        DerbyInitializer derbyInit;
+        String dbDriverName = this.testProps.getProperty("dbDriverName");
+        if (dbDriverName != null && dbDriverName.contains("derby")) {
+            derbyInit = new DerbyInitializer(this.testProps);
+            derbyInit.bootstrapDb(false);
+        }
+    }
+    
+    @Override
+    public FHIRPersistence getPersistenceImpl() throws Exception {
+        return new FHIRPersistenceJDBCImpl(this.testProps);
+    }
+}
```

Comment:
honestly, I don't fually understand why we need AbstractIncludeRevincludeTest etc and put them in the persistence project instead of the jdbc project, the same for other persistence tests, Does this mean in addition to jdbc, we can also support others?

Reply:
yes.  the persistence layer tests were designed this way because they are done more at the functional level (black box testing)...the logic in these tests should be applicable to any persistence layer implementation, not just our JDBC implementation.  we've used a few other implementations internally (Cloudant, HBase, etc), but we've only used the JDBC one in R4 and so thats the only one we've open-sourced.